test/widgets/subscription_list_test.dart

```diff
@@ -57,6 +59,23 @@ void main() {
     check(isUnpinnedHeaderInTree()).isFalse();
   });
 
+  testWidgets('link to other channels is shown', (tester) async {
+    final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+    final streams = List.generate(5, (index) => eg.stream());
+    await setupStreamListPage(tester,
+      streams: streams,
+      subscriptions: [eg.subscription(streams[1])]);
+
+    check(find.text(zulipLocalizations.browseMoreNChannels(4)).evaluate()).isNotEmpty();
+  });
+
+  testWidgets('link to other channels is not shown if server has no visible channels', (tester) async {
+    final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+    await setupStreamListPage(tester, streams: [], subscriptions: []);
+
+    check(find.text(zulipLocalizations.browseMoreNChannels(0)).evaluate()).isEmpty();
```

Comment:
Since this string is different from the one in the other test, we could have another positive case in comparison.